File: yu-ai-agent-java/yu-ai-service/src/main/java/com/yuaiagent/service/chat/model/AgentTask.java
package com.yuaiagent.service.chat.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 智能体任务实体类
 */
@Data
@TableName("agent_task")
public class AgentTask {

    /**
     * 主键ID（自增）
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 任务ID（UUID格式）
     */
    private String taskId;

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 关联会话ID，Manus 类型通过此字段与 message 表关联
     */
    private String chatId;

    /**
     * 任务描述（用户输入的任务指令）
     */
    private String message;

    /**
     * 任务状态：pending | running | completed | failed | cancelled
     */
    private String status;

    /**
     * 任务最终结果
     */
    private String result;

    /**
     * 执行步骤详情（JSON格式）
     * Python 完成时一次性回写，取消任务时清空
     */
    private String steps;

    /**
     * 是否删除：0-未删除 1-已删除
     */
    @TableLogic
    private Integer isDeleted;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

}

File: yu-ai-agent-python/app/models/db/agent_task.py
"""AgentTask ORM Model"""
from datetime import datetime

from sqlalchemy import BigInteger, String, Text, DateTime, Boolean, JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class AgentTask(Base):
    __tablename__ = "agent_task"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    task_id: Mapped[str] = mapped_column(String(36), nullable=False, index=True)
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, index=True)
    chat_id: Mapped[str | None] = mapped_column(String(36), nullable=True, index=True, comment="关联会话ID")
    message: Mapped[str] = mapped_column(Text, nullable=False, comment="任务描述")
    status: Mapped[str] = mapped_column(String(20), default="pending", comment="pending|running|completed|failed|cancelled")
    result: Mapped[str | None] = mapped_column(Text, nullable=True, comment="任务最终结果")
    steps: Mapped[list | None] = mapped_column(JSON, nullable=True, comment="执行步骤详情")
    is_deleted: Mapped[bool] = mapped_column(Boolean, default=False, comment="软删除标记")
    create_time: Mapped[datetime] = mapped_column(DateTime, default=datetime.now)
    update_time: Mapped[datetime] = mapped_column(DateTime, default=datetime.now, onupdate=datetime.now)
